.travis.yml

```diff
@@ -46,34 +63,22 @@ cache:
 git:
   depth: 5
 
-env:
-  global:
-  # Directory where tests are run from
-  - TEST_DIR=/tmp/test_dir/
-  - MODULE=autosklearn
-
-before_install:
-  - wget $MINICONDA_URL -O miniconda.sh
-  - bash miniconda.sh -b -p $HOME/miniconda
-  - export PATH="$HOME/miniconda/bin:$PATH"
-  - if [[ `which conda` ]]; then echo 'Conda installation successful'; else exit 1; fi
-  - conda create -n testenv --yes python=$PYTHON_VERSION pip wheel nose gxx_linux-64 gcc_linux-64 swig
-  - source activate testenv
+before_install: bash  ci_scripts/install_env.sh
 
 install:
+  - pip --version
   # Install general requirements the way setup.py suggests
-  - pip install pep8 codecov
-  # Temporarily pin the numpy version for travis-ci
-  - pip install "numpy<=1.14.5"
-  - cat requirements.txt | xargs -n 1 -L 1 pip install
+  - pip install nose pep8 codecov coverage flake8
+  - pip install -r requirements.txt
   # Install openml dependency for metadata generation unittest
   - pip install xmltodict requests liac-arff
   - pip install git+https://github.com/openml/openml-python@0b9009b0436fda77d9f7c701bd116aff4158d5e1 --no-deps
   - mkdir ~/.openml
   - echo "apikey = 610344db6388d9ba34f6db45a3cf71de" > ~/.openml/config
-  - pip install flake8
   # Debug output to know all exact package versions!
   - pip freeze
+  - python setup.py check -m -s
```

Comment:
Could you please explain why the following two lines are necessary?

Reply:
first, the check verify that there are no mistakes in the setup declaration so there won't be problems to upload it on PyPi
second, compile the cython code in place, some test were complaining that they do not see its tose compile modules